Updates to TB variant analysis tutorial

pvanheus opened the pull request:
This updates versions of tools mention in the M. tuberculosis variant analysis tutorial and tries to simplify the flow a bit by making clear that the Kraken2 section is optional.

It is currently a work in progress as a section on using workflows is going to be added and the embedded workflow updated. Work is anticipated to be finished by the 7th of June 2024.

hexylena commented on topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.md:
```diff
@@ -123,54 +123,59 @@ While one could examine the quality control report for each set of reads (forwar
+The next section, on looking for contamination in our data using Kraken2, takes a long time to run and can be skipped and perhaps
+run later if you prefer.
+
+{% include _includes/cyoa-choices.html option1="Skip-Kraken2" option2="Run-Kraken2" default="Skip-Kraken2" %}
```
```suggestion
{% include _includes/cyoa-choices.html option1="Skip Kraken2" option2="Run Kraken2" default="Skip Kraken2" %}
```

spaces will be replaced automatically with - when generating the div classes (and be more readable,)so  no changes needed to divs, this looks good

hexylena reviewed (commented):
looks good

hexylena commented on topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.md:
```diff
@@ -35,17 +37,17 @@ edam_ontology:
 
 Tuberculosis (TB) is an infectious disease caused by the bacterium *Mycobacterium tuberculosis*. According to the [WHO](https://www.who.int/tb/publications/global_report/en/), in 2018 there were 10.0 million new cases of TB worldwide and 1.4 million deaths due to the disease, making TB the world's most deadly infectious disease. The [publication](https://www.ncbi.nlm.nih.gov/pubmed/9634230) of th ...
-as RefSeq entry [NC_000962.3](https://www.ncbi.nlm.nih.gov/nuccore/NC_000962.3/). The genome comprises a single circular chromosome of some 4.4 megabases. The H37Rv strain that the genome was sequenced from is a long-preserved laboratory strain, originally [isolated](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2132400) from a patient in 1905 and [named](https://journals.sagepub.com/doi/abs/10.31 ...
+as RefSeq entry [NC_000962.3](https://www.ncbi.nlm.nih.gov/nuccore/NC_000962.3/). The genome comprises a single circular chromosome of some 4.4 megabases. The H37Rv strain that the genome was sequenced from is a long-preserved laboratory strain, originally [isolated](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2132400) from a patient in 1905 and [named](https://journals.sagepub.com/doi/abs/10.31 ...
```
_* looks off

hexylena commented on topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.md:
```diff
@@ -35,17 +37,17 @@ edam_ontology:
+This group of related species comprises of the [8](https://www.nature.com/articles/s41467-020-16626-6) [lineages](https://www.ncbi.nlm.nih.gov/pubmed/29456241) of human-infecting *M. tuberculosis* as well as predominantly animal-infecting species such as *M. bovis* and *M. pinnipedii*. Two other close relatives of Mtb, *M. leprae* and *M. lepromatosis* circulate between humans, causing the diseas ...
 
-Variation in the genome of *M. tuberculosis* (Mtb) is associated with changes in phenotype, for example [drug resistance](https://genomemedicine.biomedcentral.com/articles/10.1186/s13073-019-0660-8) and virulence. It is also useful for [outbreak investigation](https://www.frontiersin.org/articles/10.3389/fpubh.2019.00087/full) as the single nucleotide polymorphisms (SNPs) in a sample can be used  ...
+Variation in the genome of *M. tuberculosis* (Mtb) is associated with changes in phenotype, for example, [drug resistance](https://genomemedicine.biomedcentral.com/articles/10.1186/s13073-019-0660-8)](https://genomemedicine.biomedcentral.com/articles/10.1186/s13073-019-0660-8) and virulence. It is also useful for [outbreak investigation](https://www.frontiersin.org/articles/10.3389/fpubh.2019.000 ...
```
not sure this is right that url is here twice

hexylena commented on topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.md:
```diff
@@ -485,4 +512,49 @@ The next example is *SRR12416842* from an Indonesia [study](https://www.microbio
+
+> <hands-on-title>Analysing samples with the TB Variant Reporting workflow</hands-on-title>
+>
+> 1. **Import the Workflow** into Galaxy 
```
if the workflow is this one in the GTN, there's a snippet you can use that will give folks a two click import experience 

https://training.galaxyproject.org/training-material/news/2023/12/12/tutorial-run-wfh-ds.html

pvanheus replied:
That is for workflows in WorkflowHub.eu or Dockstore, right? As of right now the workflow is not in one of those repositories so is this snippet applicable?

hexylena replied:
there is one for grn workflows as well, see the first usage in Minerva tutorial or the assembly large genome tutorial

hexylena commented on topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.md:
```diff
@@ -390,6 +411,12 @@ A new dataset will be created in your history, containing the JBrowse interactiv
 
 You can now click on the names of the tracks to add them in, try the vcf file and gff file. You can see where the variants are located and which genes they are in. If you click on the BAM file you can zoom right in to see the read alignments for each variant if you wish.
```
you could consider adding the data/ folder and then including a snippet like we do in https://training.galaxyproject.org/training-material/topics/assembly/tutorials/chloroplast-assembly/tutorial.html#hands-on-visualise-mapped-reads (just after that hands on)

pvanheus replied:
Sounds good. Where do you get that data from - the `trackList.json` etc? By downloading an existing JBrowse dataset?

hexylena reviewed (commented):
looks good, some minor stuff, some optional upgrades but happy to merge after the link/italic is looking right

pvanheus commented:
I have added the workflow on WorkflowHub.eu and fixed up the JBrowse stuff. There is till one issue: the `>    {% snippet faqs/galaxy/workflows_run.md %}` on line 167 does not render correctly on my side.

pvanheus commented:
> I have added the workflow on WorkflowHub.eu and fixed up the JBrowse stuff. There is till one issue: the `> {% snippet faqs/galaxy/workflows_run.md %}` on line 167 does not render correctly on my side.

And this has now been fixed. Ready to review & merge!

wm75 commented on topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.md:
```diff
@@ -117,60 +119,71 @@ While one could examine the quality control report for each set of reads (forwar
 {: .hands_on}
 
-*Note:* We would normally examine our trimmed reads with **FastQC** and **MultiQC** again to see if the quality trimming has been successful, but in this tutorial we will move straight on to save time.
+*Note:* We would normally examine our trimmed reads with **FastQC** and **MultiQC** again to see if the quality trimming has been successful, but in this tutorial, we will move straight on to save time.
```
fastp also produces a (multiqc-supported) report with before and after trimming quality plots so I don't think many people would run fastqc after fastp again.

pvanheus replied:
Noted. The comment was left over from when Trimmomatic was used. FastQC is still included in this tutorial because it is a straightforward report to present for new users.

wm75 commented on topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.md:
```diff
@@ -485,4 +516,46 @@ The next example is *SRR12416842* from an Indonesia [study](https://www.microbio
+
+If you are new to collections, you can learn more about them in [this tutorial]({% link topics/galaxy-interface/tutorials/collections/tutorial.md %}).
+
+If you have followed all of the steps of this tutorial, you will have two samples, one named *004-2* and another *SRR12416842*. Create a collection of lists of pairs from these samples and call the collection `samples`.
```
> a collection of lists of pairs

I guess you mean just a "List of Dataset Pairs"? Collection of lists of pairs implies more nesting to me.
Either way, instructions how to do this could be nice without having people go through the collections tutorial.

pvanheus replied:
I have changed the wording as provided and include a snippet on making a list of pairs instead of directing people to the tutorial.

wm75 commented on topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.md:
```diff
@@ -485,4 +516,46 @@ The next example is *SRR12416842* from an Indonesia [study](https://www.microbio
+
+> <hands-on-title>Analysing samples with the TB Variant Reporting workflow</hands-on-title>
+>
+> 1. **Run the TB Variant Analysis workflow** {% icon workflow %} using the following parameters
```
For running the WF you could also use explanation like in https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/sars-cov-2-variant-discovery/tutorial.html#hands-on-import-the-workflow-for-your-data-into-galaxy

wm75 replied:
ah missed the snippet suggested by @hexylena that you put in already. That's even better.

pvanheus replied:
The snippet I included allows people to run the workflow directly on any of the usegalaxy.* servers. There are also FAQs for importing the workflow: `{% snippet faqs/galaxy/workflows_import.md %}` but is this still needed with the snippet I included?

hexylena replied:
no it's fine with just this snippet, that will let them run it on any recent galaxy they can access.

 sorry we didn't document the non-wfh version well enough and you had to upload it. I'll fix that documentation this week.

wm75 commented on topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.md:
```diff
@@ -485,4 +516,46 @@ The next example is *SRR12416842* from an Indonesia [study](https://www.microbio
+> <hands-on-title>Analysing samples with the TB Variant Reporting workflow</hands-on-title>
+>
+> 1. **Run the TB Variant Analysis workflow** {% icon workflow %} using the following parameters
+>    - {% icon param-files %} *Reads* the `samples` collection of your input reads
```
```suggestion
>    - {% icon param-collection %} *Reads* the `samples` collection of your input reads
```

wm75 commented on topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.md:
```diff
@@ -485,4 +516,46 @@ The next example is *SRR12416842* from an Indonesia [study](https://www.microbio
+> 1. **Run the TB Variant Analysis workflow** {% icon workflow %} using the following parameters
+>    - {% icon param-files %} *Reads* the `samples` collection of your input reads
+>
+>    - {% icon param-files %} *Reference Genome* the `Mycobacterium_tuberculosis_ancestral_reference.gbk` reference genome
```
```suggestion
>    - {% icon param-file %} *Reference Genome* the `Mycobacterium_tuberculosis_ancestral_reference.gbk` reference genome
```

wm75 commented on topics/variant-analysis/tutorials/tb-variant-analysis/workflows/tb-variant-analysis.ga:
```diff
@@ -1,582 +1,565 @@
     "format-version": "0.1",
-    "license": "MIT",
-    "name": "M. tuberculosis Variant Analysis tutorial",
+    "license": "AGPL-3.0-or-later",
```
isn't that a strange pick for a workflow license?

hexylena replied:
it's the one I use as well

pvanheus replied:
its a web-based resource, thus AGPL instead of GPL?

hexylena replied:
you're welcome to use any you like @pvanheus!

(I imagine to trigger the agpl behaviour it would have to be offered via an API, e.g. planemo running this, or via running in galaxy, but there the workflow is by definition accessible to the user and thus meeting the agpl. other execution systems would need to handle it appropriately as well)

wm75 replied:
I see, fine with me.
